process/transactionProcessor.go

```diff
@@ -497,6 +503,25 @@ func (tp *TransactionProcessor) computeTransactionStatus(tx *transaction.ApiTran
 	}
 }
 
+func checkIfFailedOnReturnMessage(allScrs []*transaction.ApiTransactionResult, tx *transaction.ApiTransactionResult) bool {
+	if returnMessageFailure(tx.Value, tx.ReturnMessage) {
+		return true
+	}
+
+	for _, scr := range allScrs {
+		if returnMessageFailure(scr.Value, scr.ReturnMessage) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func returnMessageFailure(value string, returnMessage string) bool {
```

Comment:
Maybe rename to `is...` / `has...` (since it returns a bool).

```
isZeroValue := value == "0"
```

Does it handle all cases / is it possible to have an empty string there? If so, e.g.

```
func isZeroAmount(amount string) bool {
	if amount == "" {
		return true
	}

	value, ok := big.NewInt(0).SetString(amount, 10)
	if ok {
		return value.Sign() == 0
	}

	return false
}
```

Reply:
updated as suggested